src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildPropertyParser.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+
+#nullable enable
+
+namespace Microsoft.DotNet.Cli.Utils;
+
+public static class MSBuildPropertyParser {
+    public static IEnumerable<(string key, string value)> ParseProperties(string input) {
+        var currentPos = 0;
+        StringBuilder currentKey = new StringBuilder();
+        StringBuilder currentValue = new StringBuilder();
+        
+        (string key, string value) EmitAndReset() {
+            var key = currentKey.ToString();
+            var value= currentValue.ToString();
+            currentKey.Clear();
+            currentValue.Clear();
+            return (key, value);
+        }
+
+        char? Peek() => currentPos < input.Length ? input[currentPos] : null;
+
+        bool TryConsume(out char? consumed) {
+            if(input.Length > currentPos) {
+                consumed = input[currentPos];
+                currentPos++;
+                return true;
```

Comment:
Can probably move the returns outside the if/else and just do `return consumed != null`